### PR TITLE
In parse config function

### PR DESCRIPTION
**Pull request by MAbraham24**
# Parse Config Function

## Description
This is a function that calls the Command Line Generator function and the Reformat YAML data function and uses them to create shell checks and gator grader command lines.

## Type of Change

- [x] Feature
- [ ] Bug fix
- [x] Documentation

## Contributors

- @MAbraham24 

## Reminder

 - All GitHub Actions should be in a passing state before any pull request is merged.
 - All PRs must be reviewed by at least one team member and one member of the Integration team!
 - Any issues this PR closes are tagged in the description!

### CODE REVIEW

**Review by ullrichd21 — APPROVED** [review]

File: gatorgrade/input/parse_config_function.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+from command_line_generator import generate_checks # Import function to generate shell and gatorgrader checks
+from in_file_path import parse_yaml_file, reformat_yml_data, add_checks_to_list # Import functions to parse and set up yaml file 
+
+def parse_config(file):
+    """Parse the input yaml file and generate specified checks
+    
+    Args:
+        file: Yaml file containing gatorgrade and shell command checks 
+    """
+    generate_checks(reformat_yaml_data(parse_yaml_file(file))) # Call previously generated function to modify file
```
Comment (tuduun) [inline gatorgrade/input/parse_config_function.py:11]: Hey Michael, I think the parse_config function needs to return a value. Moreover, the generate_checks function returns a dictionary, thus you need to assign the output to a variable or just return the output in the parse_config function.

**Review by tuduun — COMMENTED** [review]
Also, there were some linting errors.

File: gatorgrade/input/parse_config_function.py (hunk 1 above)
Comment (MAbraham24) [reply gatorgrade/input/parse_config_function.py:11]: I had discussed it with Professor Heinert and we ascertained that a return value wasn't necessary, nor was the need to assign the output to a variable based on the next steps of the project, which doesn't involve our team

Comment (tuduun) [reply gatorgrade/input/parse_config_function.py:11]: Alright, thanks for letting me know.

Comment (mariakimheinert) [reply gatorgrade/input/parse_config_function.py:11]: @MAbraham24, actually, @tuduun is correct! We must have overlooked the need for a return value during our discussion.

Comment (MAbraham24) [reply gatorgrade/input/parse_config_function.py:11]: Understood, will implement that!

**Review by Michionlion — CHANGES_REQUESTED** [review]
Hi @MAbraham24! Your work on `parse_config` looks good, but your GitHub Actions is failing because of the changes you've made in the `tests/` directory. Can you revert those changes so that your PR only shows changes for the `gatorgrade/input/parse_config_function.py`? Once that is complete and then GitHub Actions and linting pass, I think this should be good to merge.

**Comment by MAbraham24** [comment]
The change I had made that messed the testing up was that I seemed to have deleted the test directory. I restored it now, so it is no longer an error, but I'm still getting "Oh no!" return which I'm not sure how to move forward in solving.

**Comment by Michionlion** [comment]
> The change I had made that messed the testing up was that I seemed to have deleted the test directory. I restored it now, so it is no longer an error, but I'm still getting "Oh no!" return which I'm not sure how to move forward in solving.

That output is from `black`, and the "would reformat" part signifies that there are some files that `black` would reformat, if it was run -- so you should run black, and that issue will be fixed. Additionally, can you look the "[Files changed](https://github.com/GatorEducator/gatorgrade/pull/65/files)" tab in this PR and verify that every file mentioned there is something you intended to change? The [`cef2a65`](https://github.com/GatorEducator/gatorgrade/pull/65/commits/cef2a65cc0b802d6767434ab9021bd46bf857e46) commit added a significant amount of code that may have been intended to be in another PR.

**Review by LiamBlack3 — APPROVED** [review]
These changes look great! good work!

**Review by Michionlion — APPROVED** [review]
This looks great! I think it's good to merge!

**Pull request merged**